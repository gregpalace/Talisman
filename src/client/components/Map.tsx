import React from 'react';

const Map = (props: any) => {
  return (
    <div
    style={{
      width: '800px',
      height: '400px',
      backgroundColor: 'black',
      border: '4px solid black',
      margin: '10px auto',
    }}
    />
  );
}


export default Map;