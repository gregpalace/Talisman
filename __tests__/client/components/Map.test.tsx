import React from 'react';
import {shallow} from 'enzyme';
import toJson from 'enzyme-to-json';

import Map from '../../../src/client/components/map';

describe('React unit tests', () => {
  let wrapper: any;

  beforeEach(() => {
    wrapper = shallow(<Map />);
  });

  describe('<Map />', () => {
    it('Should render correctly', () => {
      expect(toJson(wrapper)).toMatchSnapshot();
    });
    it('Should redner a wrapper with style properties', () => {
      expect(wrapper.find('style')).toBeTruthy();
      expect(wrapper.prop('style').width).toEqual('800px');
      expect(wrapper.prop('style').height).toEqual('400px');
      expect(wrapper.prop('style').backgroundColor).toEqual('black');
    })
  });
});