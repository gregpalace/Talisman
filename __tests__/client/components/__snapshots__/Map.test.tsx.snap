// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`React unit tests <Map /> Should render correctly 1`] = `
<div
  style={
    Object {
      "backgroundColor": "black",
      "border": "4px solid black",
      "height": "400px",
      "margin": "10px auto",
      "width": "800px",
    }
  }
/>
`;
